### ISSUE TITLE
'dmnorm_chol' not recognized as nimble function

### ISSUE THREAD

**Issue by eslawler** [nimble-dev/nimble#1037]
When trying to use the nimble function 'dmnorm_chol' in building a model, I continue to get the following message and error:

> defining model...
Registering the following user-provided distributions: dmnorm_chol
Error in FUN(X[[i]], ...) : 
  checkDistributionFunctions: density function for dmnorm_chol is not available.  It must be a nimbleFunction (with no setup code).

Code that produced the problem: 
```
library(nimble)
library(coda)
library(MASS)

## simulate dataset
p <- 10    # number of predictors
n <- 100   # number of observations
X <- matrix(rnorm(p*n), nrow = n, ncol = p) # design matrix
true_betas <- seq(0.1, 1, length.out = p) # "true" parameters
true_sigma <- diag(rep(1,n)) # "true" covariance matrix
y <- mvrnorm(1, X %*% true_betas, true_sigma) # observations

## build model
# prior and likelihood distributions
mvn_code <- nimbleCode({
  # priors on beta and scalar for covariance matrix
  for(i in 1:p) beta[i] ~ dnorm(0, sd=100)
  sigma_scalar ~ dinvgamma(shape = 0.01, scale = 0.1)
  
  # covariance matrix
  sigma <- chol(sigma_scalar * diag(rep(1,n)))
  
  # likelihood
  for(j in 1:n) {
    y[j] ~ dmnorm_chol(inprod(beta[1:p], x[j, 1:p]), sigma, prec_param = FALSE)
  }
})

X <- sweep(X, 2, colMeans(X)) 
constants <- list(n = n, p = p, x = X)
data <- list(y = y)
inits <- list(beta = rep(0, p), sigma_scalar = 0.5)
mvn_model <- nimbleModel(mvn_code, constants = constants, data = data, inits = inits)
```

Upon execution of the last line ("mvn_model"), I receive the error message included above. I don't believe I should be receiving this error when using a nimble function (which 'dmnorm_chol' is).

OS, R version, and nimble package version:
- Darwin Kernel Version 19.5.0: Tue May 26 20:41:44 PDT 2020; root:xnu-6153.121.2~2/RELEASE_X86_64 x86_64
- R version 4.0.1
- nimble version 0.9.1

Any help/insight you could provide would be much appreciated -- thanks in advance!

**Comment by paciorek** [comment]
There's a subtle distinction here between using the Cholesky version of `dmnorm` in a model versus in a nimbleFunction. 

If you want to use the Cholesky version of `dmnorm` in a model, you simply specify:

```
y[1:n] ~ dmnorm(mu[1:n], cholesky = U[1:n,1:n], prec_param = 0)
```

in your model code. 

You can see more details on different parameterizations of `dmnorm` that are available in the manual in Section 5.2.4.2.

Use of `dmnorm_chol` is only if used directly in a nimbleFunction.

**Comment by eslawler** [comment]
Thank you for the quick reply! I appreciate you explaining that distinction and how to go about implementing this in a model.
